angular-client/src/pages/charging-page/components/fault-display/fault-display.component.ts

```diff
@@ -12,37 +11,41 @@ export default class FaultDisplay {
   faults: { faultName: string; time: string }[] = [];
   resetButton = {
     onClick: () => {
-      this.faults = [];
+      this.faults.shift();
     },
     icon: 'restart_alt'
   };
   constructor(private storage: Storage) {}
 
   ngOnInit() {
     this.storage.get(IdentifierDataType.COMM_TIMEOUT_FAULT).subscribe((value) => {
-      this.faultPushCheck(floatPipe(value.values[0]));
+      this.faultPushCheck(value.values[0], 'Comm Timeout');
     });
 
     this.storage.get(IdentifierDataType.HARDWARE_FAILURE_FAULT).subscribe((value) => {
-      this.faultPushCheck(floatPipe(value.values[0]));
+      this.faultPushCheck(value.values[0], 'Hardware Failure');
     });
 
     this.storage.get(IdentifierDataType.OVER_TEMP_FAULT).subscribe((value) => {
-      this.faultPushCheck(floatPipe(value.values[0]));
+      this.faultPushCheck(value.values[0], 'Over Temp');
     });
 
     this.storage.get(IdentifierDataType.VOLTAGE_WRONG_FAULT).subscribe((value) => {
-      this.faultPushCheck(floatPipe(value.values[0]));
+      this.faultPushCheck(value.values[0], 'Voltage Wrong');
     });
 
     this.storage.get(IdentifierDataType.WRONG_BAT_CONNECT_FAULT).subscribe((value) => {
-      this.faultPushCheck(floatPipe(value.values[0]));
+      this.faultPushCheck(value.values[0], 'Wrong Battery Connect');
     });
   }
 
-  faultPushCheck(numberValue: number) {
-    if (numberValue !== 0 && this.faults.length <= 50) {
-      this.faults.push({ faultName: 'Wrong Battery Connect', time: new Date().toLocaleTimeString() });
+  // Takes in a string (which should be an integer) representing the storage value of a fault, and the faults name.
+  // If the string is 0 their is no fault, anything else means there was a fault.
+  // (this is check is required because the storage service still takes information from the server even if a fault hasn't happened, requiring
+  // us to ensure that change to the storage value was actually a fault)
```

Comment:
can you format this comment in JDoc format, also you shouldn't need lines 44/45, line 43 makes it clear why we're throwing out some values